src/ripple/app/main/NodeStoreScheduler.cpp

```diff
@@ -47,11 +47,14 @@ void NodeStoreScheduler::onChildrenStopped ()
 
 void NodeStoreScheduler::scheduleTask (NodeStore::Task& task)
 {
-    ++m_taskCount;
-    m_jobQueue->addJob (
-        jtWRITE,
-        "NodeObject::store",
-        [this, &task] (Job&) { doTask(task); });
+    if (! isStopping())
+    {
+        ++m_taskCount;
+        m_jobQueue->addJob (
+            jtWRITE,
+            "NodeObject::store",
+            [this, &task] (Job&) { doTask(task); });
+    }
```

Comment:
I felt the same way. But when rippled shuts down, it vital that we write dirty SLEs belonging to validated ledgers. So it would be a mistake to try to short circuit that.